pandas/tests/arrays/test_integer.py

```diff
@@ -361,22 +378,22 @@ def _compare_other(self, data, op_name, other):
 
         # array
         result = pd.Series(op(data, other))
-        expected = pd.Series(op(data._data, other))
+        expected = pd.Series(op(data._data, other), dtype="boolean")
 
         # fill the nan locations
-        expected[data._mask] = op_name == "__ne__"
+        expected[data._mask] = pd.NA
 
         tm.assert_series_equal(result, expected)
```

Comment:
In https://github.com/pandas-dev/pandas/pull/29961 I have found this TestComparisonOps tests not fully covering. Eg, I think you will also want to test comparing with a scalar `pd.NA`